stroyprombeton/views/catalog.py

```diff
@@ -11,15 +11,19 @@
 from images.models import Image
 from pages.models import CustomPage, ModelPage
 from pages.templatetags.pages_extras import breadcrumbs as get_page_breadcrumbs
-from stroyprombeton import context as stb_context, models, request_data
+from stroyprombeton import context as stb_context, models, exception, request_data
 from stroyprombeton.views.helpers import set_csrf_cookie
 
 
 def fetch_products(request):
     """Filter product table on Category page by Name, code, specification."""
-    context_ = stb_context.FetchPositions(
-        request_data.FetchProducts(request, url_kwargs={}),
-    )
+    try:
+        context_ = stb_context.FetchPositions(
+            request_data.FetchProducts(request, url_kwargs={}),
+        )
+    except exception.Http400 as e:
+        return HttpResponseBadRequest(str(e))
```

Comment:
Let's create a middleware to catch our custom exceptions. It'd be more convenient, robust